app/scripts/views/sign_in_recovery_code.js

```diff
@@ -0,0 +1,70 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+const AuthErrors = require('../lib/auth-errors');
+const Cocktail = require('cocktail');
+const FormView = require('./form');
+const SignInMixin = require('./mixins/signin-mixin');
+const ServiceMixin = require('./mixins/service-mixin');
+const Template = require('templates/sign_in_recovery_code.mustache');
+const VerificationReasonMixin = require('./mixins/verification-reason-mixin');
+
+const CODE_INPUT_SELECTOR = 'input.recovery-code';
+const VALID_RECOVERY_CODE_REGEX = /^([a-zA-Z0-9]{10})|([a-fA-F0-9]{8})/;
```

Comment:
You'll want to add `$` on the end of the regexp, otherwise this will just check to ensure the code is a minimum of 8 characters long instead of exactly 8 or 10 characters.